### PR TITLE
Validate shared views

### LINKED ISSUE #20950: Validate compatibility and uniquify view columns
And views that clash after being truncated

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #20950

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
@msftbot Require 1 sign off

File: src/EFCore.Relational/Extensions/RelationalForeignKeyExtensions.cs [hunk 1]
```diff
@@ -101,8 +89,8 @@ public static string GetDefaultName([NotNull] this IForeignKey foreignKey)
                 var linkedForeignKey = rootForeignKey.DeclaringEntityType
                     .FindRowInternalForeignKeys(storeObject)
                     .SelectMany(fk => fk.PrincipalEntityType.GetForeignKeys())
-                    .FirstOrDefault(k => principalTableName == k.PrincipalEntityType.GetTableName()
-                        && principalSchema == k.PrincipalEntityType.GetSchema()
+                    .FirstOrDefault(k => principalStoreObject.Name == k.PrincipalEntityType.GetTableName()
+                        && principalStoreObject.Schema == k.PrincipalEntityType.GetSchema()
```
Comment (smitpatel) [inline src/EFCore.Relational/Extensions/RelationalForeignKeyExtensions.cs:93]: Do we also need to check that principalStoreObject is of type table.

File: src/EFCore.Relational/Infrastructure/RelationalModelValidator.cs [hunk 2]
```diff
@@ -439,6 +439,145 @@ static bool IsNotNullAndFalse(object value)
...
+        /// <summary>
+        ///     Validates the compatibility of entity types sharing a given view.
+        /// </summary>
+        /// <param name="mappedTypes"> The mapped entity types. </param>
+        /// <param name="viewName"> The view name. </param>
+        /// <param name="schema"> The schema. </param>
+        /// <param name="logger"> The logger to use. </param>
+        protected virtual void ValidateSharedViewCompatibility(
+            [NotNull] IReadOnlyList<IEntityType> mappedTypes,
+            [NotNull] string viewName,
```
Comment (smitpatel) [inline src/EFCore.Relational/Infrastructure/RelationalModelValidator.cs:488]: can pass storeobject here.

File: src/EFCore.Relational/Extensions/RelationalForeignKeyExtensions.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Extensions/RelationalForeignKeyExtensions.cs:93]: We only create FKs for tables, so whatever happens for non-tables doesn't matter

File: src/EFCore.Relational/Infrastructure/RelationalModelValidator.cs (hunk 2 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Infrastructure/RelationalModelValidator.cs:488]: I wanted to keep `ValidateSharedViewCompatibility` and `ValidateSharedTableCompatibility` as separate methods as the overlap is small. When separate storeobject would contain redundant information.

**Pull request merged**